include/pybind11/functional.h

```diff
@@ -50,9 +50,9 @@ struct type_caster<std::function<Return(Args...)>> {
         }
 
         auto src = reinterpret_borrow<object>(src_);
-        value = [src](Args... args) -> Return {
+        value = [src](Args&&... args) -> Return {
```

Comment:
Ah, thank you.  I hadn't caught that they weren't being deduced here.  The `&&` has now been removed.